src/lib/components/Tabs/index.js

```diff
@@ -94,8 +96,119 @@ class Tabs extends BaseElement {
     `;
   }
 
+  panelTemplate(i, child) {
+    return html`
+      <div
+        id="web-tab-${this.idSalt}-${i}-panel"
+        class="web-tabs__panel"
+        role="tabpanel"
+        aria-labelledby="web-tab-${this.idSalt}-${i}"
+        hidden
+      >
+        ${child}
+      </div>
+    `;
+  }
+
+  firstUpdated() {
+    this.activeTab = 0;
+    this.classList.remove("unresolved");
+    this.onResize();
+
+    // If Tabs component contains AssessmentQuestion components,
+    // listen for requests to navigate to the next tab.
+    const questions = this.querySelectorAll("web-question");
+
+    if (!questions) return;
+    for (const question of questions) {
+      question.addEventListener("request-nav-to-next", this.nextTab);
+    }
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener("resize", this.onResize);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    window.removeEventListener("resize", this.onResize);
+  }
+
+  set activeTab(val) {
+    const oldVal = this.activeTab_;
+
+    this.activeTab_ = Math.floor(val);
```

Comment:
I don't think this is worth a getter/setter. I think the idiomatic way to do this is to remove all this and let LitElement implicitly call `requestUpdate` itself.

Then, [overload `updated`](https://lit-element.polymer-project.org/guide/lifecycle#updated):

```js
  updated(changedProperties) {
    if (changedProperties.has("activeTab") {
      this._changeTab();  // _ prefix because it's an internal method
    }
  }
```

.... and either don't worry about floating-point `activeTab`, or use `Math.floor` when you use the value.

Reply:
Yeah, sorry, there's lots of dumb stuff in this PR. This was my first attempt at writing a web component...